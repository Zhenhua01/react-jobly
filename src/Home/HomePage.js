import { useNavigate } from "react-router-dom";

/**Presentational Component:
 * HomePage for JoblyApp
 *
 * RoutesList -> HomePage
 */
function HomePage() {
  const navigate = useNavigate();

  function loginClick() {
    navigate("/login");
  }

  function signupClick() {
    navigate("/signup");
  }

  return (
    <div className='HomePage'>
      <h1>Jobly</h1>
      <p>All the jobs in one, convenient place.</p>
      <button onClick={loginClick} className="btn-primary btn Login-addBtn">
        Login
      </button>
      <button onClick={signupClick} className="btn-primary btn Signup-addBtn">
        Signup
      </button>
    </div>
  );
}

export default HomePage;