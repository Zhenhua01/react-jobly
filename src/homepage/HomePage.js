import { useNavigate } from "react-router-dom";
import { useContext } from "react";
import userContext from '../context/userContext';


/** HomePage for Jobly App
 *
 * Context: user
 *  { username, firstName, lastName, email, isAdmin, applications:[] }
 *
 * JoblyApp -> RoutesList -> HomePage
 */

function HomePage() {
  const { user } = useContext(userContext);
  const navigate = useNavigate();

  // Handles login click.
  function loginClick() {
    navigate("/login");
  }

  // Handles signup click.
  function signupClick() {
    navigate("/signup");
  }

  return (
    <div className="HomePage">
      <div className="container-fluid text-center">
        <h1 className="mt-3 mb-3">Jobly</h1>

        {user &&
          <>
            <h4>Welcome back {user.firstName}.</h4>
          </>
        }

        {!user &&
          <>
            <h4 className="mb-4">All the jobs in one, convenient place.</h4>
            <button type="button"
              onClick={loginClick}
              className="btn btn-lg btn-info fw-bold me-3">
              Login
            </button>
            <button type="button"
              onClick={signupClick}
              className="btn btn-lg btn-info fw-bold">
              Signup
            </button>
          </>
        }
      </div>
    </div>
  );
}

export default HomePage;