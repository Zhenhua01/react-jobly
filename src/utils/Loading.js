import React from "react";
import "./Loading.css";


/** Loading message to display when making API calls for data. */

function Loading() {
  return (
    <div className="Loading">
      Loading ...
    </div>
  );
}

export default Loading;