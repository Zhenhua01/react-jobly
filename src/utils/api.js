import axios from "axios";

// Use port 3001 when connecting to local backend server
// const BASE_URL = process.env.REACT_APP_BASE_URL || "http://localhost:3001";

// Use heroku deployed backend when local server not available
const BASE_URL =
    process.env.REACT_APP_BASE_URL || "https://jobly-zhl-exp.herokuapp.com";

/** API Class.
 *
 * Static class tying together methods used to get/send to to the API.
 * There shouldn't be any frontend-specific stuff here, and there shouldn't
 * be any API-aware stuff elsewhere in the frontend.
 */

 class JoblyApi {

  static token = "";

  static async request(endpoint, data = {}, method = "get") {
    console.debug("API Call:", endpoint, data, method);

    const url = `${BASE_URL}/${endpoint}`;
    const headers = { Authorization: `Bearer ${JoblyApi.token}` };
    const params = (method === "get")
        ? data
        : {};

    try {
      return (await axios({ url, method, data, params, headers })).data;
    } catch (err) {
      console.error("API Error:", err.response);
      let message = err.response.data.error.message;
      throw Array.isArray(message) ? message : [message];
    }
  }

  // Individual API routes

  /** Get details on a company by handle. */

  static async getCompany(handle) {
    let res = await this.request(`companies/${handle}`);
    return res.company;
  }

  /** Get all companies or companies matching search term. */

  static async getCompanies(data = {}) {
    let res = await this.request(`companies`, data);
    return res.companies;
  }

  /** Get all jobs or jobs matching search term. */

  static async getJobs(data = {}) {
    let res = await this.request(`jobs`, data);
    return res.jobs;
  }

  /** Registers a new user. */

  static async signup(data) {
    let res = await this.request(`auth/register`, data, "post");
    return res.token;
  }

  /** Logins user. */

  static async login(data) {
    let res = await this.request(`auth/token`, data, "post");
    return res.token;
  }

  /** Update user information. */

  static async update(username, data) {
    let res = await this.request(`users/${username}`, data, "patch");
    return res.user;
  }

  /**Get user information. */

  static async getUserInfo(username) {
    let res = await this.request(`users/${username}`);
    return res.user;
  }

  /**Add job to user applications. */

  static async applyToJob(username, id) {
    await this.request(`users/${username}/jobs/${id}`, {}, "post");
    let res = await this.request(`users/${username}`);
    return res.user;
  }
}


export default JoblyApi;