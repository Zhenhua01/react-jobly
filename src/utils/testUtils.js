import React from "react";
import userContext from "../context/userContext";

const demoUser = {
  username: "testuser",
  first_name: "testfirst",
  last_name: "testlast",
  email: "test@test.net",
  photo_url: null,
  applications: [],
};

const UserProvider = ({
  children,
  user = demoUser,
  hasAppliedToJob = () => false,
}) => (
  <userContext.Provider value={{ user, hasAppliedToJob }}>
    {children}
  </userContext.Provider>
);

export { UserProvider };
