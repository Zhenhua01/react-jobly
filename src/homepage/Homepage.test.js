import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router";
import HomePage from "./Homepage";
import { UserProvider } from "../utils/testUtils";


it("matches snapshot", function () {
  const { asFragment } = render(
      <MemoryRouter>
        <UserProvider>
          <HomePage />
        </UserProvider>
      </MemoryRouter>,
  );
  expect(asFragment()).toMatchSnapshot();
});

it("matches snapshot when logged out", function () {
  const { asFragment } = render(
      <MemoryRouter>
        <UserProvider user={null}>
          <HomePage />
        </UserProvider>
      </MemoryRouter>,
  );
  expect(asFragment()).toMatchSnapshot();
});
