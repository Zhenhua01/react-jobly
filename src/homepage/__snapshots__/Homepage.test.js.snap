// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="HomePage"
  >
    <div
      class="container-fluid text-center"
    >
      <h1
        class="mt-3 mb-3"
      >
        Jobly
      </h1>
      <h4>
        Welcome back 
      </h4>
    </div>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot when logged out 1`] = `
<DocumentFragment>
  <div
    class="HomePage"
  >
    <div
      class="container-fluid text-center"
    >
      <h1
        class="mt-3 mb-3"
      >
        Jobly
      </h1>
      <h4
        class="mb-4"
      >
        All the jobs in one, convenient place.
      </h4>
      <button
        class="btn btn-lg btn-info fw-bold me-3"
        type="button"
      >
        Login
      </button>
      <button
        class="btn btn-lg btn-info fw-bold"
        type="button"
      >
        Signup
      </button>
    </div>
  </div>
</DocumentFragment>
`;
